src/LIBSVM.jl

```diff
@@ -106,7 +106,13 @@ function SVM(smc::SVMModel, y, X, weights, labels, svmtype, kernel)
         unsafe_copyto!(pointer(libsvmweight_label), smc.param.weight_label, nw)
     end
 
-    SVM(svmtype, kernel, weights, size(X,1),
+    if kernel == Kernel.Precomputed
```

Comment:
Not sure why we need this?
In case of `Kernel.Precomputed`, the `size(X)` is `(l, l)`. So simply taking `size(X, 1)` is fine for me.

Reply:
Ostensibly, yes. But after [Line 337](https://github.com/JuliaML/LIBSVM.jl/blob/683cfbb925bf4c30207c4ba05ae4e65f125fa8a0/src/LIBSVM.jl#L337) `X` will be a row vector with indices only, i.e. of dimension `(1,l)`. 

For what it's worth, this is something that is addressed in the other pull request (see the discussion [here](https://github.com/JuliaML/LIBSVM.jl/pull/88#discussion_r744743812)) out of necessity, but in the interest of keeping this one self-contained & simple I didn't make any changes to the code that aren't strictly necessary.